main.py

```diff
@@ -22,6 +22,7 @@
 from segment_anything.modeling import Sam
 from groundingdino.models import GroundingDINO
 
+ID_FROM_PHRASE = {}
```

Comment:
Why did you need to make this a global variable? Too much work to pass around?

Reply:
Felt like the easiest way, but I changed it since it's not too much work to avoid here.